scripts/fix-daily-branch

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+# Fix daily build branches so daily build recipes by dropping all 'cherry-pick'
+# quilt patches applied to a given ubuntu/* release branch.
+#  Any *cpick* files in debian/patches are git revert'd in reverse order
+# so that the upstream commits the *cpick* file represent are able to apply.
+set -e
+
+error() { echo "$@" 1>&2; }
+fail() { [ $# -eq 0 ] || error "$@"; exit 1; }
+
+gitcmd() {
+    git "$@" || fail "Failed running git $@"
+}
+
+Usage() {
+cat <<EOF
+${0##*/}
+
+Create a daily_branch which reverts any debian/patches/*cpick* commits from
+source_branch.
+
+options:
+   -h|--help                   show this message.
+   --source-branch          The local branch name from which to revert all
+                            debian/patches/*cpick* files. Default: ubuntu/devel.
+   --daily-branch           The local daily branch name to create which reverts
+                            debian/patches/*cpick* files. Default:
+                            ubuntu/daily/devel
+EOF
+}
+
+drop_cherry_picks_from_daily_branch() {
+    local source_branch="${1}" daily_branch="${2}" daily_remote="${3:-origin}"
+    gitcmd fetch ${daily_remote}
+    echo "Backing out all debian/patches/*cpick* from ${daily_branch} branch"
+    if git branch | grep -q ${daily_branch}; then
+        gitcmd checkout ${daily_branch}
+    else
+        # Create local daily branch from current
+        gitcmd checkout ${source_branch} -b ${daily_branch}
+    fi
+    gitcmd reset --hard ${source_branch}
+    gitcmd log --oneline -- debian/patches/*cpick* | cut -d" " -f1 | xargs git revert
+    # switch back to current development branch
+    gitcmd checkout ${source_branch}
+    cat <<EOF
+To fix daily recipe builds perform the following:
+
+  # Verify local changes in $daily_branch versus $source_branch
+  git diff $daily_branch
+  git push $daily_remote $daily_branch
+EOF
+}
+
+main() {
+    local short_opts="hd:s:r:"
+    local long_opts="help,daily-branch:,source_branch:,remote:"
+    local getopt_out=""
+    getopt_out=$(getopt --name "${0##*/}" \
+        --options "${short_opts}" --long "${long_opts}" -- "$@") &&
+        eval set -- "${getopt_out}" ||
+        { Usage; return; }
+
+    local cur="" next="" daily_branch="ubuntu/daily/devel"
+    local source_branch="ubuntu/devel" remote="origin"
+
+    while [ $# -ne 0 ]; do
+        cur="$1"; next="$2";
+        case "$cur" in
+            -h|--help) Usage ; exit 0;;
+            -d|--daily-branch) daily_branch=$next; shift;;
+            -r|--remote) remote=$next; shift;;
+            -s|--source-branch) source_branch=$next; shift;;
+            --) shift; break;;
+        esac
+        shift;
+    done
+
+    drop_cherry_picks_from_daily_branch $source_branch $daily_branch $remote
```

Comment:
Do we need to check that $source_branch and $daily_branch have actually been set?